src/status_im2/contexts/communities/discover/view.cljs

```diff
@@ -104,47 +118,126 @@
+
+(defn render-sticky-header
+  [selected-tab]
+  (fn [scroll-height]
+    (when (> scroll-height 348)
+      [react/blur-view
+       {:blur-amount   32
+        :blur-type     :xlight
+        :overlay-color (if platform/ios? colors/white-opa-70 :transparent)
+        :style         style/blur-tabs-header}
+       [discover-communities-segments selected-tab]])))
+
+(defn discover-screen-content
+  [featured-communities]
+  (let [view-type (reagent/atom :card-view)
+        selected-tab (reagent/atom :all)
+        featured-communities-count (count featured-communities)
+        scroll-component (scroll-page/scroll-page
+                          {:background-color  (colors/theme-colors
+                                               colors/white
+                                               colors/neutral-95)
+                           :content-container (fn []
+                                                [rn/view {:margin-top 112}
+                                                 [discover-communities-header
+                                                  {:featured-communities-count featured-communities-count
+                                                   :featured-communities       featured-communities
+                                                   :view-type                  @view-type
+                                                   :selected-tab               selected-tab}]])
+                           :name              (i18n/label :t/discover-communities)})]
+    (fn []
+      (let [sticky-header  (memoize (render-sticky-header selected-tab))
+            page-component (memoize (render-page-content selected-tab @view-type))]
+        (fn []
+          (scroll-component
+           sticky-header
+           page-component))))))
 
 (defn discover
   []
-  (let [view-type (reagent/atom :card-view)]
-    (fn []
-      (let [communities                (rf/sub [:communities/sorted-communities])
-            featured-communities       (rf/sub [:communities/featured-communities])
-            featured-communities-count (count featured-communities)]
-        [safe-area/consumer
-         (fn []
-           [rn/view
-            {:style {:margin-left      20
-                     :margin-right     20
-                     :flex             1
-                     :background-color (colors/theme-colors
-                                        colors/white
-                                        colors/neutral-90)}}
-            [quo/button
-             {:icon     true
-              :type     :grey
-              :size     32
-              :style    {:margin-vertical 12}
-              :on-press #(rf/dispatch [:navigate-back])}
-             :i/close]
-            [screen-title]
-            [featured-communities-header featured-communities-count]
-            [featured-list featured-communities @view-type]
-            [other-communities-list communities @view-type]])]))))
+  (fn []
```

Comment:
This component doesn't need to be a form-2 component, i.e. you can remove the surrounding anonymous function.